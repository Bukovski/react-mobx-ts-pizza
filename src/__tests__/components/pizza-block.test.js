import React from 'react';
import { render, screen } from "@testing-library/react";
import PizzaBlock from "../../components/pizza-block.component";
import { mockPizza } from "../../__mocks__/db.mock";
import userEvent from "@testing-library/user-event";


describe("PizzaBlock Component", () => {
	it("should render with defaultProps", () => {
		const { container } = render(<PizzaBlock />);
		
		expect(screen.getByText("---")).toHaveClass("pizza-block__title");
		expect(screen.getByText("тонкое")).toHaveClass("disabled");
		expect(screen.getByText("традиционное")).toHaveClass("disabled");
		expect(screen.getByText("от 0 ₽")).toHaveClass("pizza-block__price");

		expect(container.firstChild).toMatchSnapshot()
	});
	
	it("should render with props", () => {
		render(<PizzaBlock
			{ ...mockPizza.pizzas[ 1 ] }
			addedCount={ 3 }
		/>);
		
		expect(screen.getByText("Сырная")).toHaveClass("pizza-block__title");
		
		expect(screen.getByText("тонкое")).toHaveClass("active");
		expect(screen.getByText("традиционное")).toHaveClass("disabled");
		
		expect(screen.getByText("26 см.").closest('li')).toHaveClass("active");
		expect(screen.getByText("30 см.").closest('li')).toHaveClass("disabled");
		
		expect(screen.getByText("от 245 ₽")).toHaveClass("pizza-block__price");
		
		expect(screen.getByText(3).closest('button')).toBeInTheDocument();
	});
	
	it("should call onClickAddPizza function if click button", () => {
		const mockClickAddPizza = jest.fn();
		
		render(<PizzaBlock
			{ ...mockPizza.pizzas[ 1 ] }
			addedCount={ 3 }
			onClickAddPizza={ mockClickAddPizza }
		/>);
		
		expect(screen.getByText("Сырная")).toHaveClass("pizza-block__title");
		
		expect(mockClickAddPizza).toBeCalledTimes(0);
		
		userEvent.click(screen.getByRole("button"))
		expect(mockClickAddPizza).toBeCalledTimes(1);
	});
	
	it("should select size & types", () => {
		const mockClickAddPizza = jest.fn();
		
		render(<PizzaBlock
			{ ...mockPizza.pizzas[ 0 ] }
			addedCount={ 3 }
			onClickAddPizza={ mockClickAddPizza }
		/>);
		
		expect(screen.getByText("тонкое")).toHaveClass("active");
		expect(screen.getByText("традиционное")).not.toHaveClass("active");

		userEvent.click(screen.getByText("традиционное"))
		expect(screen.getByText("тонкое")).not.toHaveClass("active");
		expect(screen.getByText("традиционное")).toHaveClass("active");
		
		
		expect(screen.getByText("26 см.").closest('li')).toHaveClass("active");
		expect(screen.getByText("30 см.").closest('li')).not.toHaveClass("active");
		
		userEvent.click(screen.getByText("30 см."))
		expect(screen.getByText("26 см.").closest('li')).not.toHaveClass("active");
		expect(screen.getByText("30 см.").closest('li')).toHaveClass("active");
		
		expect(mockClickAddPizza).toBeCalledTimes(0);
	});
});
