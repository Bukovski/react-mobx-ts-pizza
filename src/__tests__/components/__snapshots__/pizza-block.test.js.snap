// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PizzaBlock Component should render with defaultProps 1`] = `
<section
  class="pizza-block"
  data-testid="pizza-block"
>
  <img
    alt="Pizza"
    class="pizza-block__image"
  />
  <h4
    class="pizza-block__title"
  >
    ---
  </h4>
  <div
    class="pizza-block__selector"
  >
    <ul>
      <li
        class="disabled"
      >
        тонкое
      </li>
      <li
        class="disabled"
      >
        традиционное
      </li>
    </ul>
    <ul>
      <li
        class="active disabled"
      >
        26
         см.
      </li>
      <li
        class="disabled"
      >
        30
         см.
      </li>
      <li
        class="disabled"
      >
        40
         см.
      </li>
    </ul>
  </div>
  <div
    class="pizza-block__bottom"
  >
    <div
      class="pizza-block__price"
    >
      от 
      0
       ₽
    </div>
    <button
      class="button button--add button--outline"
    >
      <svg
        fill="none"
        height="12"
        viewBox="0 0 12 12"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.8 4.8H7.2V1.2C7.2 0.5373 6.6627 0 6 0C5.3373 0 4.8 0.5373 4.8 1.2V4.8H1.2C0.5373 4.8 0 5.3373 0 6C0 6.6627 0.5373 7.2 1.2 7.2H4.8V10.8C4.8 11.4627 5.3373 12 6 12C6.6627 12 7.2 11.4627 7.2 10.8V7.2H10.8C11.4627 7.2 12 6.6627 12 6C12 5.3373 11.4627 4.8 10.8 4.8Z"
          fill="white"
        />
      </svg>
      <span>
        Добавить
      </span>
      <i />
    </button>
  </div>
</section>
`;
